src/main/java/com/mongodb/kafka/connect/source/StartedMongoSourceTask.java

```diff
@@ -0,0 +1,649 @@
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import com.mongodb.lang.Nullable;
+
+import com.mongodb.kafka.connect.source.heartbeat.HeartbeatManager;
+import com.mongodb.kafka.connect.source.producer.SchemaAndValueProducer;
+import com.mongodb.kafka.connect.source.statistics.StatisticsManager;
+import com.mongodb.kafka.connect.source.topic.mapping.TopicMapper;
+import com.mongodb.kafka.connect.util.VisibleForTesting;
+import com.mongodb.kafka.connect.util.jmx.SourceTaskStatistics;
+import com.mongodb.kafka.connect.util.time.InnerOuterTimer;
+import com.mongodb.kafka.connect.util.time.InnerOuterTimer.InnerTimer;
+
+final class StartedMongoSourceTask implements AutoCloseable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoSourceTask.class);
+  private static final String FULL_DOCUMENT = "fullDocument";
+  private static final int NAMESPACE_NOT_FOUND_ERROR = 26;
+  private static final int ILLEGAL_OPERATION_ERROR = 20;
+  private static final int INVALIDATED_RESUME_TOKEN_ERROR = 260;
+  private static final int CHANGE_STREAM_FATAL_ERROR = 280;
+  private static final int CHANGE_STREAM_HISTORY_LOST = 286;
+  private static final int BSON_OBJECT_TOO_LARGE = 10334;
+  private static final Set<Integer> INVALID_CHANGE_STREAM_ERRORS =
+      new HashSet<>(
+          asList(
+              INVALIDATED_RESUME_TOKEN_ERROR,
+              CHANGE_STREAM_FATAL_ERROR,
+              CHANGE_STREAM_HISTORY_LOST,
+              BSON_OBJECT_TOO_LARGE));
+  private static final String RESUME_TOKEN = "resume token";
+  private static final String RESUME_POINT = "resume point";
+  private static final String NOT_FOUND = "not found";
+  private static final String DOES_NOT_EXIST = "does not exist";
+  private static final String INVALID_RESUME_TOKEN = "invalid resume token";
+  private static final String NO_LONGER_IN_THE_OPLOG = "no longer be in the oplog";
+
+  private final Supplier<SourceTaskContext> sourceTaskContextAccessor;
+  private final Time time;
+  private volatile boolean isRunning;
+  private boolean isCopying;
+
+  private final MongoSourceConfig sourceConfig;
+  private final Map<String, Object> partitionMap;
+  private final MongoClient mongoClient;
+  private HeartbeatManager heartbeatManager;
+
+  private boolean supportsStartAfter = true;
+  private boolean invalidatedCursor = false;
+  @Nullable private final MongoCopyDataManager copyDataManager;
+  private BsonDocument cachedResult;
+  private BsonDocument cachedResumeToken;
+
+  private MongoChangeStreamCursor<? extends BsonDocument> cursor;
+  private final StatisticsManager statisticsManager;
+  private final InnerOuterTimer inTaskPollInConnectFrameworkTimer;
+
+  StartedMongoSourceTask(
+      final Supplier<SourceTaskContext> sourceTaskContextAccessor,
+      final MongoSourceConfig sourceConfig,
+      final MongoClient mongoClient,
+      @Nullable final MongoCopyDataManager copyDataManager,
+      final StatisticsManager statisticsManager) {
```

Comment:
`StartedMongoSourceTask` was extracted from `MongoSourceTask`, it gets all its dependencies via a constructor, which makes it unit-testable.